src/main/java/de/tum/in/www1/artemis/domain/notification/SingleUserNotificationFactory.java

```diff
@@ -115,4 +119,82 @@ public static SingleUserNotification createNotification(PlagiarismCase plagiaris
+                notification.setTransientAndStringTarget(createTutorialGroupTarget(tutorialGroup, tutorialGroup.getCourse().getId(), true, true));
+            }
+            case TUTORIAL_GROUP_DEREGISTRATION_TUTOR -> {
+                if (Objects.isNull(tutorialGroup.getTeachingAssistant())) {
+                    throw new IllegalArgumentException("The tutorial group " + tutorialGroup.getTitle() + " does not have a tutor to which a notification could be sent.");
+                }
+
+                var student = users.stream().findAny();
+                var studentName = student.isPresent() ? student.get().getName() : "";
+
+                notification = new SingleUserNotification(tutorialGroup.getTeachingAssistant(), title, "The student " + studentName
+                        + " has been deregistered from your tutorial group " + tutorialGroup.getTitle() + " by " + responsibleForAction.getName() + ".");
+                notification.setTransientAndStringTarget(createTutorialGroupTarget(tutorialGroup, tutorialGroup.getCourse().getId(), true, true));
+            }
+            case TUTORIAL_GROUP_MULTIPLE_REGISTRATION_TUTOR -> {
+                if (Objects.isNull(tutorialGroup.getTeachingAssistant())) {
+                    throw new IllegalArgumentException("The tutorial group " + tutorialGroup.getTitle() + " does not have a tutor to which a notification could be sent.");
+                }
+                notification = new SingleUserNotification(tutorialGroup.getTeachingAssistant(), title,
+                        users.size() + " students have been registered to your tutorial group " + tutorialGroup.getTitle() + " by " + responsibleForAction.getName() + ".");
+
+                notification.setTransientAndStringTarget(createTutorialGroupTarget(tutorialGroup, tutorialGroup.getCourse().getId(), true, true));
+            }
+            case TUTORIAL_GROUP_ASSIGNED -> {
+                var tutorToContact = users.stream().findAny().get();
+                notification = new SingleUserNotification(tutorToContact, title,
+                        "You have been assigned to lead the tutorial group " + tutorialGroup.getTitle() + " by " + responsibleForAction.getName() + ".");
+                notification.setTransientAndStringTarget(createTutorialGroupTarget(tutorialGroup, tutorialGroup.getCourse().getId(), true, true));
+            }
+            case TUTORIAL_GROUP_UNASSIGNED -> {
+                var tutorToContact = users.stream().findAny().get();
+                notification = new SingleUserNotification(tutorToContact, title,
+                        "You have been unassigned from leading the tutorial group " + tutorialGroup.getTitle() + " by " + responsibleForAction.getName() + ".");
+                notification.setTransientAndStringTarget(createTutorialGroupTarget(tutorialGroup, tutorialGroup.getCourse().getId(), true, true));
```

Comment:
Q: This notification currently creates a link to `https://artemis-test1.artemis.in.tum.de/course-management/24/tutorial-groups-management/18`, which might not make sense when I was unassigned from leading the group (do I still have permission as normal tutor)?

Reply:
Yeah. You can still see the detail page. But the edit buttons and such will be deactivated